aggregators/codec.go

```diff
@@ -64,7 +64,11 @@ func (k *CombinedMetricsKey) SizeBinary() int {
 // ToProto converts CombinedMetrics to its protobuf representation.
 func (m *CombinedMetrics) ToProto() *aggregationpb.CombinedMetrics {
 	pb := aggregationpb.CombinedMetricsFromVTPool()
-	pb.ServiceMetrics = make([]*aggregationpb.KeyedServiceMetrics, 0, len(m.Services))
+	if pb.ServiceMetrics == nil {
+		pb.ServiceMetrics = make([]*aggregationpb.KeyedServiceMetrics, 0, len(m.Services))
+	} else {
+		pb.ServiceMetrics = pb.ServiceMetrics[:0]
+	}
```

Comment:
This can lead to extra allocations than before if the capacity of `pb.ServiceMetrics` is less than `len(m.Services)`

Reply:
True, but I think we are doing something similar in APM to reuse Batch iirc. I don't think there's a good way around this unless we pool by size. Any suggestions?

Reply:
Sounds reasonable. Will we end up with a large amount of large slices in the pool, though less likely over time?